features/vaadin-opennms-pluginmanager/src/main/java/org/opennms/features/pluginmgr/vaadin/config/opennms/internal/HttpServletRequestVaadinImpl.java

```diff
@@ -0,0 +1,446 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2013-2014 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2014 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.pluginmgr.vaadin.config.opennms.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.security.Principal;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.servlet.AsyncContext;
+import javax.servlet.DispatcherType;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.Part;
+
+import org.opennms.web.api.OnmsHeaderProvider;
+
+import com.vaadin.server.VaadinRequest;
+
+/**
+ * 
+ * This class creates an {@link HttpServletRequest} object that delegates all calls to
+ * a {@link VaadinRequest} instance. This is used so that we can fetch the header HTML
+ * from an {@link OnmsHeaderProvider}.
+ *
+ * TODO: Refactor into a common class.
+ * TODO: this is a copy of org.opennms.features.topology.app.internal.HttpServletRequestVaadinImpl
+ */
+@Deprecated
+public class HttpServletRequestVaadinImpl implements HttpServletRequest {
```

Comment:
Do we need this? The `VaadinRequest` on initializiation of the Vaadin Application is a `VaadinServletRequest`.